src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCreateCollectorFileServiceImpl.java

```diff
@@ -1,50 +1,126 @@
+import org.kuali.kfs.sys.KFSPropertyConstants;
 import org.kuali.kfs.sys.service.UniversityDateService;
 import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.krad.bo.BusinessObject;
 
+import edu.cornell.kfs.concur.ConcurConstants;
 import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractFile;
 import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractCreateCollectorFileService;
 import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractValidationService;
 import edu.cornell.kfs.sys.CUKFSConstants;
 import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.CUKFSPropertyConstants;
+import edu.cornell.kfs.sys.batch.service.BusinessObjectFlatFileSerializerService;
 
+@SuppressWarnings("deprecation")
 public class ConcurStandardAccountingExtractCreateCollectorFileServiceImpl
         implements ConcurStandardAccountingExtractCreateCollectorFileService {
 
     private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(
             ConcurStandardAccountingExtractCreateCollectorFileServiceImpl.class);
 
+    protected static final String DATE_RANGE_FORMAT = "%s..%s";
+    protected static final int DEFAULT_BUILDER_SIZE = 100;
+
     protected ConcurStandardAccountingExtractValidationService concurSAEValidationService;
+    protected BusinessObjectFlatFileSerializerService collectorFlatFileSerializerService;
+    protected LookupableHelperService batchFileLookupableHelperService;
     protected UniversityDateService universityDateService;
     protected DateTimeService dateTimeService;
     protected ParameterService parameterService;
+    protected String collectorDirectoryPath;
 
     @Override
-    public boolean buildCollectorFile(ConcurStandardAccountingExtractFile saeFileContents) {
-        ConcurStandardAccountingExtractCollectorBatchBuilder builder = new ConcurStandardAccountingExtractCollectorBatchBuilder(
-                universityDateService, dateTimeService, concurSAEValidationService, this::getConcurParameterValueAsString);
+    public String buildCollectorFile(ConcurStandardAccountingExtractFile saeFileContents) {
+        ConcurStandardAccountingExtractCollectorBatchBuilder builder = createBatchBuilder();
         
-        CollectorBatch collectorBatch = builder.buildCollectorBatchFromStandardAccountingExtract(0, saeFileContents);
+        int sequenceNumber = calculateBatchSequenceNumber();
+        CollectorBatch collectorBatch = builder.buildCollectorBatchFromStandardAccountingExtract(sequenceNumber, saeFileContents);
         if (collectorBatch == null) {
-            LOG.error("There was a problem preparing the data for the Collector file; will not create a file.");
-            return false;
+            LOG.error("There was a problem preparing the data for the Collector file; will not create a file. See earlier logs for details.");
+            return null;

```

Comment:
This probably should be reworked so that we are not returning nulls in this method.